fhir-persistence/src/main/java/com/ibm/fhir/persistence/ResourceResult.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * (C) Copyright IBM Corp. 2021, 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.ibm.fhir.model.annotation.Required;
+import com.ibm.fhir.model.resource.Resource;
+
+/**
+ * The base wrapper result wrapper used to represent a resource being returned from a
+ * persistence interaction.
+ * Instances are immutable and can be constructed via {@code new ResourceResult.Builder<T>()}.
+ */
+public class ResourceResult<T extends Resource> {
+    @Required
+    private final T resource;
+    private final boolean deleted;
+    private final String resourceTypeName;
+    private final String logicalId;
+    private final int version;
+    private final Instant lastUpdated;
+
+    /**
+     * Private constructor used by the Builder to create a new ResourceResult instance
+     * @param builder
+     */
+    private ResourceResult(Builder<T> builder) {
+        resource = builder.resource;
+        deleted = builder.deleted;
+        resourceTypeName = builder.resourceTypeName;
+        logicalId = builder.logicalId;
+        version = builder.version;
+        lastUpdated = builder.lastUpdated;
+    }
+
+    /**
+     * Create a Builder for building instances of this class
+     * @param <T>
+     * @return
+     */
+    public static <T extends Resource> ResourceResult.Builder<T> builder() {
+        return new ResourceResult.Builder<>();
+    }
+
+    /**
+     * Convenience method to convert a Resource value to a ResourceResult,
+     * which is useful for unit-tests
+     *
+     * @param <T>
```

Comment:
super minor
```suggestion
     * @param <T>
```

Reply:
Interesting...this gets added by Eclipse. Removed.